src/db.js

```diff
@@ -0,0 +1,273 @@
+    console.log(dailycount);
+    return dailycount <= dailyLimit;
+
+}
+
+async function resetDailyCount(addressDetails){
+    const now = new Date();
+    // console.log(addressDetails);
+    const address = addressDetails.address;
+    const dailytime = addressDetails.dailytime;
+    if ((Math.floor(now.getTime()/1000 - Math.floor(dailytime.getTime()/1000))) > 86400){
+        //update
+        console.log('Resetting...');
+        const update = 'update depositortest set dailycount=0,dailytime=$1 where address= $2 returning dailycount'
+        const values = [now,address]
+        const dailycount = await pool.query(update,values);
+        return dailycount.rows[0].dailycount; //daily limit has been reset
+    }
+    return addressDetails.dailycount;
+    
+
+}
+
+async function checkWeeklyLimit(addressDetails){
+    const weeklycount = await resetWeeklyCount(addressDetails);
+    return weeklycount <= weeklyLimit;
+}
+
+async function resetWeeklyCount(addressDetails){
+    const now = new Date();
+    const address = addressDetails.address;
+    const weeklytime = addressDetails.weeklytime;
+
+    if ((Math.floor(now.getTime()/1000 - Math.floor(weeklytime.getTime()/1000))) > 604800){
+        //update
+        const update = 'update depositortest set weeklycount=0,weeklytime=$1 where address= $2 returning weeklycount'
+        const values = [now,address]
+        const weeklycount = await pool.query(update,values);
+        return weeklycount.rows[0].weeklycount; //weekly limit has been reset
+    }
+    return addressDetails.weeklycount;
+}
+
+async function resetNoRequests(addressDetails){
+    // console.log(addressDetails)
+    const now = new Date();
+    const address = addressDetails.address;
+    const firstrequesttime = addressDetails.firstrequesttime;
+    if ((Math.floor(now.getTime()/1000 - Math.floor(firstrequesttime.getTime()/1000))) > 172800){
+        //update
+        const update = 'update depositortest set norequests=0,firstrequesttime=$1 where address= $2 returning norequests'
+        const values = [now,address]
+        const norequests = await pool.query(update,values);
+        return norequests.rows[0].norequests; //daily limit has been reset
+    }
+    return addressDetails.norequests;
+}
+
+async function updateCounts(addressDetails,topUpAmount){
+    var newDailyCount = addressDetails.dailycount + topUpAmount;
+    var newWeeklyCount = addressDetails.weeklycount + topUpAmount;
+    
+    const update = 'update depositortest set dailycount= $1,weeklycount= $2 where address= $3';
+    const values = [newDailyCount,newWeeklyCount, addressDetails.address];
+    await pool.query(update,values);
+
+}
+  
+async function objectRowUpdate(addressDetails){
+      console.log('Inside object row:', addressDetails);
+      const update = 'update depositortest set validatedtx= $1,unaccountedamount= $2, unaccountedtx= $3, dailycount= $4, weeklycount= $5 where address= $6';
```

Comment:
Check all params here to to be escaped and have exactly expected types when used in query